qa/ccs-common-rest/src/yamlRestTest/java/org/elasticsearch/test/rest/yaml/RcsCcsCommonYamlTestSuiteIT.java

```diff
@@ -294,20 +293,9 @@ protected ClientYamlTestExecutionContext createRestTestExecutionContext(
                 getClusterStateFeatures(adminSearchClient),
                 semanticNodeVersions
             );
-            final TestFeatureService combinedTestFeatureService = new TestFeatureService() {
-                @Override
-                public boolean clusterHasFeature(String featureId) {
-                    return testFeatureService.clusterHasFeature(featureId) && searchTestFeatureService.clusterHasFeature(featureId);
-                }
+            final TestFeatureService combinedTestFeatureService = featureId -> testFeatureService.clusterHasFeature(featureId)
```

Comment:
if `TestFeatureService` extended `Predicate<String>`, we could use `Predicate#and` here

Reply:
I agree, it would be nice, but we'd have to go with Predicate<String> only instead of TestFeatureService